app/controllers/posts_controller.rb

```diff
@@ -1,30 +1,43 @@
 class PostsController < ApplicationController
+  load_and_authorize_resource :user
+  load_and_authorize_resource :post, through: :user, shallow: true
   def index
-    @user = User.find(params[:user_id])
+    @user = User.includes(:posts).find(params[:user_id])
     @posts = @user.posts
   end
 
   def show
     @user = User.find(params[:user_id])
     @post = @user.posts.find(params[:id])
-    @comments = @post.comments
+    @comments = Comment.where(post_id: params[:id])
   end
 
   def create
     @post = current_user.posts.new(post_params)
 
     if @post.save
       redirect_to user_posts_path(current_user)
+
     else
       flash[:alert] = 'Something went wrong'
-      render :new
+      render 'new'
     end
   end
 
   def new
     @post = Post.new
   end
 
+  def include_user
+    @posts = @user.posts.includes(:comments)
+  end
+
+  def destroy
+    @post.destroy
+    @post.likes.destroy_all
+    redirect_to user_posts_path(@user)
+  end
```

Comment:
- [ ] Great job implementing authorization rules. However, there's little bug here. I am unable to delete a post with comments. Below is the error I got while trying to do just that. Kindly look into it and fix.

### Error
![destroy-error](https://github.com/atok624/Blog-app-rails/assets/26597432/fe26e237-4baf-4992-97be-05a50bb3ff36)

Reply:
We have resolved that
Thank you